http-cors/src/main/scala/org/apache/pekko/http/cors/scaladsl/settings/CorsSettings.scala

```diff
@@ -0,0 +1,236 @@
+  // See https://github.com/akka/akka-http/issues/661 for a potential opening of the JavaMapping API.
+  override def withAllowGenericHttpRequests(newValue: Boolean): CorsSettings = {
+    copy(allowGenericHttpRequests = newValue)
+  }
+  override def withAllowCredentials(newValue: Boolean): CorsSettings = {
+    copy(allowCredentials = newValue)
+  }
+  override def withAllowedOrigins(newValue: javadsl.model.HttpOriginMatcher): CorsSettings = {
+    copy(allowedOrigins = newValue.asInstanceOf[HttpOriginMatcher])
+  }
+  override def withAllowedHeaders(newValue: javadsl.model.HttpHeaderRange): CorsSettings = {
+    copy(allowedHeaders = newValue.asInstanceOf[HttpHeaderRange])
+  }
+  override def withAllowedMethods(
+      newValue: java.lang.Iterable[org.apache.pekko.http.javadsl.model.HttpMethod]): CorsSettings = {
+    copy(allowedMethods = newValue.asScala.toList.asInstanceOf[List[HttpMethod]])
+  }
+  override def withExposedHeaders(newValue: java.lang.Iterable[String]): CorsSettings = {
+    copy(exposedHeaders = newValue.asScala.toList)
+  }
+  override def withMaxAge(newValue: OptionalLong): CorsSettings = {
+    copy(maxAge = newValue.toScala)
+  }
+
+  // overloads for Scala idiomatic use
+  def withAllowedOrigins(newValue: HttpOriginMatcher): CorsSettings = copy(allowedOrigins = newValue)
+  def withAllowedHeaders(newValue: HttpHeaderRange): CorsSettings = copy(allowedHeaders = newValue)
+  def withAllowedMethods(newValue: Seq[HttpMethod]): CorsSettings = copy(allowedMethods = newValue)
+  def withExposedHeaders(newValue: Seq[String]): CorsSettings = copy(exposedHeaders = newValue)
+  def withMaxAge(newValue: Option[Long]): CorsSettings = copy(maxAge = newValue)
+
+  private[pekko] def preflightResponseHeaders(origins: Seq[HttpOrigin], requestHeaders: Seq[String]): List[HttpHeader]
+  private[pekko] def actualResponseHeaders(origins: Seq[HttpOrigin]): List[HttpHeader]
+}
+
+object CorsSettings {
+  private val prefix = "pekko.http.cors"
+  final private val MaxCached = 8
+  private[this] var cache = ListMap.empty[ActorSystem, CorsSettings]
+
+  /**
+   * Creates an instance of settings using the given Config.
+   */
+  def apply(config: Config): CorsSettings =
+    fromSubConfig(config.getConfig(prefix))
+
+  /**
+   * Creates an instance of settings using the given String of config overrides to override settings set in the class
+   * loader of this class (i.e. by application.conf or reference.conf files in the class loader of this class).
+   */
+  def apply(configOverrides: String): CorsSettings =
+    apply(ConfigFactory.parseString(configOverrides).withFallback(ConfigFactory.load(getClass.getClassLoader)))
+
+  /**
+   * Creates an instance of CorsSettings using the configuration provided by the given ActorSystem.
+   */
+  def apply(system: ActorSystem): CorsSettings =
+    // From private pekko.http.impl.util.SettingsCompanionImpl implementation
```

Comment:
The `SettingsCompanionImpl` could now be used, to remove the copy-pasted code

Reply:
Fixed in latest push